TikTacToeGame/app/src/main/java/com/r4ziel/tiktactoegame/GameFragmentViewModel.kt

```diff
@@ -68,44 +101,51 @@ class GameFragmentViewModel : ViewModel() {
+        savedState[IS_GAME_IN_PROGRESS_KEY] = isGameInProgress
+        savedState[PLAYER_1_LIST_KEY] = player1BlockList
+        savedState[PLAYER_2_LIST_KEY] = player2BlockList
+        savedState[PLAYER_TURN_KEY] = playerTurn
+        savedState[DRAW_GAME_LIST_KEY] = drawGameBlockList
+        savedState[IS_DRAW_KEY] = isDrawLiveData.value
+        savedState[IS_GAME_OVER_KEY] = isGameOverLiveData.value
+        savedState[BLOCK_LIST_KEY] = blockLiveData.value
+    }
+
     /**
      * Methods Determining Win Lose Or Draw
      */
 
     fun isDrawGame(): Boolean {
-        return drawGameBlockList.contains(1) && drawGameBlockList.contains(2) && drawGameBlockList.contains(3) &&
-            drawGameBlockList.contains(4) && drawGameBlockList.contains(5) && drawGameBlockList.contains(6) &&
-            drawGameBlockList.contains(7) && drawGameBlockList.contains(8) && drawGameBlockList.contains(9)
+        return drawGameBlockList.size == 9 && !isGameOver()
     }
 
     fun isGameOver(): Boolean {
 
-        if ((player1BlockList.contains(1) && player1BlockList.contains(2) && player1BlockList.contains(3)) ||
-            (player1BlockList.contains(4) && player1BlockList.contains(5) && player1BlockList.contains(6)) ||
-            (player1BlockList.contains(7) && player1BlockList.contains(8) && player1BlockList.contains(9)) ||
-            (player1BlockList.contains(1) && player1BlockList.contains(4) && player1BlockList.contains(7)) ||
-            (player1BlockList.contains(2) && player1BlockList.contains(5) && player1BlockList.contains(8)) ||
-            (player1BlockList.contains(3) && player1BlockList.contains(6) && player1BlockList.contains(9)) ||
-            (player1BlockList.contains(1) && player1BlockList.contains(5) && player1BlockList.contains(9)) ||
-            (player1BlockList.contains(3) && player1BlockList.contains(5) && player1BlockList.contains(7))) {
-
-            winner = 1
-            return true
-        } else if ((player2BlockList.contains(1) && player2BlockList.contains(2) && player2BlockList.contains(3)) ||
-            (player2BlockList.contains(4) && player2BlockList.contains(5) && player2BlockList.contains(6)) ||
-            (player2BlockList.contains(7) && player2BlockList.contains(8) && player2BlockList.contains(9)) ||
-            (player2BlockList.contains(1) && player2BlockList.contains(4) && player2BlockList.contains(7)) ||
-            (player2BlockList.contains(2) && player2BlockList.contains(5) && player2BlockList.contains(8)) ||
-            (player2BlockList.contains(3) && player2BlockList.contains(6) && player2BlockList.contains(9)) ||
-            (player2BlockList.contains(1) && player2BlockList.contains(5) && player2BlockList.contains(9)) ||
-            (player2BlockList.contains(3) && player2BlockList.contains(5) && player2BlockList.contains(7))) {
-
-            winner = 2
-            return true
+        winner = if (player1BlockList.size == player2BlockList.size) {
+            2
```

Comment:
You can simplify this further:
```suggestion
        winner = 1
        if (player1BlockList.size == player2BlockList.size) winner = 2
```